diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1e29fd1..bff9b7d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -248,6 +248,10 @@ macro(copy_file IN_FILE OUT_DIR)
     endif()
 endmacro()
 
+find_package(slang CONFIG REQUIRED)
+target_link_libraries(slang-rhi PUBLIC slang::slang)
+
+if(0)
 # Fetch slang
 if(SLANG_RHI_FETCH_SLANG)
     set(SLANG_VERSION ${SLANG_RHI_FETCH_SLANG_VERSION})
@@ -330,7 +334,7 @@ if(NOT SLANG_RHI_BUILD_FROM_SLANG_REPO)
         copy_file(${SLANG_RHI_SLANG_BINARY_DIR}/bin/slang-rt.dll .)
     endif()
 endif()
-
+endif()
 # Slang distributes prelude headers in the "include" directory along the other headers.
 # However, in a slang build tree, the files are not in the "include" directory but in a separate "prelude" directory.
 # Make sure to include the prelude directory if it exists.
@@ -431,7 +435,7 @@ if(SLANG_RHI_ENABLE_OPTIX)
 endif()
 
 # Fetch & setup Vulkan headers
-if(SLANG_RHI_ENABLE_VULKAN)
+if(0)
     message(STATUS "Fetching Vulkan headers ...")
     FetchPackage(vulkan_headers URL ${SLANG_RHI_VULKAN_HEADERS_URL})
     add_library(slang-rhi-vulkan-headers INTERFACE)
@@ -498,7 +502,10 @@ configure_file(
     ${CMAKE_CURRENT_SOURCE_DIR}/include/slang-rhi-config.h.in
     ${CMAKE_CURRENT_BINARY_DIR}/include/slang-rhi-config.h
 )
-target_include_directories(slang-rhi PUBLIC ${CMAKE_CURRENT_BINARY_DIR}/include)
+target_include_directories(slang-rhi PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+)
 
 if(APPLE)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-assume -Wno-switch")
@@ -560,12 +567,12 @@ target_compile_options(slang-rhi-warnings INTERFACE
     >
 )
 add_library(slang-rhi-warnings-as-errors INTERFACE)
-target_compile_options(slang-rhi-warnings-as-errors INTERFACE
-    $<$<CXX_COMPILER_ID:MSVC>:/WX>
-    $<$<CXX_COMPILER_ID:GNU>:-Werror>
-    $<$<CXX_COMPILER_ID:Clang>:-Werror>
-    $<$<CXX_COMPILER_ID:AppleClang>:-Werror>
-)
+# target_compile_options(slang-rhi-warnings-as-errors INTERFACE
+#     $<$<CXX_COMPILER_ID:MSVC>:/WX>
+#     $<$<CXX_COMPILER_ID:GNU>:-Werror>
+#     $<$<CXX_COMPILER_ID:Clang>:-Werror>
+#     $<$<CXX_COMPILER_ID:AppleClang>:-Werror>
+# )
 target_link_libraries(slang-rhi PRIVATE slang-rhi-warnings slang-rhi-warnings-as-errors)
 
 target_sources(slang-rhi PRIVATE
@@ -702,7 +709,9 @@ if(SLANG_RHI_ENABLE_VULKAN)
         src/vulkan/vk-texture.cpp
         src/vulkan/vk-utils.cpp
     )
-    target_link_libraries(slang-rhi PRIVATE slang-rhi-vulkan-headers)
+
+    find_package( VulkanHeaders CONFIG REQUIRED )
+    target_link_libraries(slang-rhi PUBLIC Vulkan::Headers)
 endif()
 
 if(SLANG_RHI_ENABLE_METAL)
@@ -783,7 +792,12 @@ if(SLANG_RHI_ENABLE_WGPU)
     )
 endif()
 
-target_include_directories(slang-rhi PUBLIC include)
+target_include_directories(slang-rhi
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+)
+
 target_include_directories(slang-rhi PRIVATE src)
 target_compile_definitions(slang-rhi
     PUBLIC
@@ -985,8 +999,13 @@ endif()
 # Install
 if(SLANG_RHI_INSTALL)
     install(
-        TARGETS slang-rhi
+        TARGETS
+            slang-rhi
+            slang-rhi-resources cmrc-base
+            slang-rhi-warnings slang-rhi-warnings-as-errors
+        EXPORT slang-rhi-targets
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     )
     install(
@@ -998,4 +1017,28 @@ if(SLANG_RHI_INSTALL)
         FILES ${CMAKE_CURRENT_BINARY_DIR}/include/slang-rhi-config.h
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
     )
+    install(EXPORT slang-rhi-targets
+        FILE slang-rhi-targets.cmake
+        NAMESPACE slang::
+        DESTINATION share/slang-rhi
+    )
+    file(CONFIGURE
+        OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/slang-rhi-config.cmake"
+        CONTENT
+            [=[
+            include(CMakeFindDependencyMacro)
+
+            find_dependency(slang CONFIG)
+
+            if(@SLANG_RHI_ENABLE_VULKAN@)
+                find_dependency(VulkanHeaders CONFIG)
+            endif()
+
+            include("${CMAKE_CURRENT_LIST_DIR}/slang-rhi-targets.cmake")
+            ]=]
+        @ONLY
+    )
+    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/slang-rhi-config.cmake"
+            DESTINATION share/slang-rhi
+    )
 endif()
diff --git a/include/slang-rhi.h b/include/slang-rhi.h
index 2ff4aa6..f73d8b6 100644
--- a/include/slang-rhi.h
+++ b/include/slang-rhi.h
@@ -19,6 +19,7 @@
 #endif
 #endif
 #else // SLANG_RHI_SHARED
+#error "sss"
 #define SLANG_RHI_API
 #endif // SLANG_RHI_SHARED
 
